src/xchain/modules/folks-gmp.ts

```diff
@@ -296,4 +355,13 @@ export const util = {
     );
     return await this.spokeToHubMessageFee(adapterId, startFolksChainId, undefined, receiverValue, gasLimit);
   },
+
+  async calcHubToSpokeMsgValue(msgValueEstimationArgs: MsgValueEstimationArgs): Promise<bigint> {
+    const { folksChainId, accountId, poolId, returnAdapterId, returnGasLimit } = msgValueEstimationArgs;
+    const folksTokenId = getFolksTokenIdFromPool(poolId);
+    const bridgeRouterValue = await util.bridgeRouterHubBalance(accountId);
+    const estimatedValue = await util.hubToSpokeMessageFee(returnAdapterId, folksChainId, folksTokenId, returnGasLimit);
```

Comment:
Can do these calls in parallel

Reply:
Method deleted 842694f2970aa711b6473283e007a5968f8eb416